src/io/iohk/scalanet/peergroup/TCPPeerGroup.scala

```diff
@@ -206,15 +203,17 @@ object TCPPeerGroup {
       messageSubject.onComplete()
 
     override def channelRead(ctx: ChannelHandlerContext, msg: Any): Unit = {
-
-      val messageE: Either[Codec.Failure, M] = codec.decode(msg.asInstanceOf[ByteBuf].nioBuffer().asReadOnlyBuffer())
-      log.debug(
-        s"Processing inbound message from remote address ${ctx.channel().remoteAddress()} " +
-          s"to local address ${ctx.channel().localAddress()}, ${messageE.getOrElse("decode failed")}"
-      )
-
-      messageE.foreach { message =>
-        messageSubject.onNext(message)
+      val byteBuf = msg.asInstanceOf[ByteBuf]
```

Comment:
Does `msg: ByteBuf` work? if it does, please remove the casting.

Reply:
msg type is `Any`, so cast seems necessary

Reply:
It could be worth to do pattern matching instead and throw a custom exception when the cast fails.